### PR TITLE
BUG: fix test failures on aarch64 and other architectures

### PR DESCRIPTION
**Pull request by drakenclimber**
This commit is intended to address Issue #210 where tests 53 and 55 are failing on non-x86_64 architectures.  Both tests now utilize actual syscall names rather than fictitious syscall numbers.

### CODE REVIEW

File: src/arch-syscall-dump.c [hunk 1]
```diff
@@ -127,6 +128,9 @@ int main(int argc, char *argv[])
 		case SCMP_ARCH_PPC64LE:
 			sys = ppc64_syscall_iterate(iter);
 			break;
+		case SCMP_ARCH_RISCV64:
+			sys = riscv64_syscall_iterate(iter);
+			break;
```
Comment (pcmoore) [inline src/arch-syscall-dump.c]: All the changes in `arch-syscall-dump.c` probably should be split out into a separate patch.  Yes, it is a bug fix, but it is a *different* bug that you are fixing.

File: tests/53-sim-binary_tree.c [hunk 2]
```diff
@@ -49,38 +93,26 @@ int main(int argc, char *argv[])
...
+	for (i = 0; i < table_size; i++) {
 		/* arbitrarily make the filter more complex by filtering
 		 * on arguments for a few syscalls
 		 */
-		if (i == 10 || i == 53 || i == 61 || i == 255)
-			rc = seccomp_rule_add(ctx, SCMP_ACT_ERRNO(i), i, 1,
-					      SCMP_A0(SCMP_CMP_EQ, i));
+		if (table[i].syscall == SCMP_SYS(mprotect) ||
+		    table[i].syscall == SCMP_SYS(dup) ||
+		    table[i].syscall == SCMP_SYS(alarm))
```
Comment (pcmoore) [inline tests/53-sim-binary_tree.c]: Why not add an additional field to the syscall table to indicate that this syscall should have an argument added to the filter?

File: tests/53-sim-binary_tree.c [hunk 3]
```diff
@@ -29,9 +29,53 @@
 
 #include "util.h"
 
-#define MAX_SYSCALL		(330)
+struct syscall_errno {
+	int syscall;
+	int error;
```
Comment (pcmoore) [inline tests/53-sim-binary_tree.c]: See my comments below about adding an additional field to signify that a given entry should add an argument to the filter.

File: tests/53-sim-binary_tree.py [hunk 4]
```diff
@@ -36,11 +77,15 @@ def test(args):
...
     f.set_attr(Attr.CTL_OPTIMIZE, 2)
 
-    for i in range(330):
-        if (i == 10 or i == 53 or i == 61 or i == 255):
-            f.add_rule(ERRNO(i), i, Arg(0, EQ, i))
+    for entry in table:
+        print(entry)
+        if entry["syscall"] == "mprotect" or \
+           entry["syscall"] == "dup" or \
+           entry["syscall"] == "alarm":
```
Comment (pcmoore) [inline tests/53-sim-binary_tree.py]: My comments from the C variant of this test still apply here although the implementation will obviously be slightly different.

File: tests/55-basic-pfc_binary_tree.c [hunk 5]
```diff
@@ -51,31 +76,31 @@ int main(int argc, char *argv[])
...
+	for (i = 0; i < table_size; i++) {
 		/* arbitrarily make the filter more complex by filtering
 		 * on arguments for a few syscalls
 		 */
-		if (i == 10 || i == 53 || i == 61 || i == 255)
-			rc = seccomp_rule_add(ctx, SCMP_ACT_ERRNO(i), i, 1,
-					      SCMP_A0(SCMP_CMP_EQ, i));
+		if (table[i].syscall == SCMP_SYS(write) ||
+		    table[i].syscall == SCMP_SYS(open) ||
+		    table[i].syscall == SCMP_SYS(pread64))
```
Comment (pcmoore) [inline tests/55-basic-pfc_binary_tree.c]: The test 53 comments also apply here.

File: src/arch-syscall-dump.c (hunk 1 above)
Comment (drakenclimber) [reply src/arch-syscall-dump.c]: Good plan.  It looks like we missed RISCV support in a few of the tools.  I'll make a separate pull request for them soon

Comment (drakenclimber) [reply src/arch-syscall-dump.c]: Addressed in pull request #212

**Comment by pcmoore** [comment]
Merged via HEAD 3fb32e45dbde214d7bd05da51fd0a9d6abfb8265, thanks!
